### PR TITLE
Integrate blocks api

### PR DESCRIPTION
**Pull request by kansi**
## Description
Integrate the [blocks api](https://docs.bigchaindb.com/projects/server/en/latest/http-client-server-api.html#get--api-v1-blocks-block_id) with Tendermint.

## Related issues
#1828
 #1946

## Impacted Areas in Application
This PR breaks compatiability with existing [`/api/v1/blocks/{block_id}`](https://docs.bigchaindb.com/projects/server/en/latest/http-client-server-api.html#get--api-v1-blocks-block_id), see #1946

### CODE REVIEW

File: bigchaindb/models.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def from_dict(cls, tx_body):
         return super().from_dict(tx_body)
 
     @classmethod
-    def from_db(cls, bigchain, tx_dict):
+    def from_db(cls, bigchain, tx_dict_list):
```
Comment (ttmc) [inline bigchaindb/models.py]: Please update the docstring as well.

File: bigchaindb/tendermint/lib.py [hunk 2]
```diff
@@ -111,6 +111,47 @@ def get_latest_block(self):
...
         return backend.query.get_latest_block(self.connection)
 
+    def get_block(self, block_id, include_status=False):
+        """Get the block with the specified `block_id` (and optionally its status)
+
+        Returns the block corresponding to `block_id` or None if no match is
+        found.
+
+        Args:
+            block_id (str): transaction id of the transaction to get
```
Comment (ttmc) [inline bigchaindb/tendermint/lib.py]: I think you meant, "block ID of the block to get"

File: docs/server/source/data-models/block-model.rst [hunk 3]
```diff
@@ -25,66 +19,10 @@ The JSON Keys in a Block
...
-It's a string.
-To compute it, 1) construct an :term:`associative array` ``d`` containing
-``block.timestamp``, ``block.transactions``, ``block.node_pubkey``,
-``block.voters``, and their values. 2) compute ``id = hash_of_aa(d)``.
-There's pseudocode for the ``hash_of_aa()`` function
-in the `IPDB Transaction Spec page about cryptographic hashes
-<https://the-ipdb-transaction-spec.readthedocs.io/en/latest/common-operations/crypto-hashes.html#computing-the-hash-of-an-associative-array>`_.
-The result (``id``) is a string: the block ID.
-An example is ``"b60adf655932bf47ef58c0bfb2dd276d4795b94346b36cbb477e10d7eb02cea8"``
+The block ID denotes the height of the blockchain when the given block was committed.
```
Comment (ttmc) [inline docs/server/source/data-models/block-model.rst]: Please add that `id` is an integer (or is it a string?) and that it starts at 0 (or does it start at 1?).

File: docs/server/source/http-client-server-api.rst [hunk 4]
```diff
@@ -583,12 +583,10 @@ Blocks
...
 .. http:get:: /api/v1/blocks/{block_id}
 
-   Get the block with the ID ``block_id``. Any blocks, be they ``VALID``, ``UNDECIDED`` or ``INVALID`` will be
-   returned. To check a block's status independently, use the `Statuses endpoint <#status>`_.
-   To check the votes on a block, have a look at the `votes endpoint <#votes>`_.
+   Get the block with the ID ``block_id``.
 
    :param block_id: block ID
-   :type block_id: hex string
+   :type block_id: string
```
Comment (ttmc) [inline docs/server/source/http-client-server-api.rst]: The "Example request" and "Example response" also need to be updated. Here's a screenshot of what the rendered docs look like for this branch:

screenshot from 2017-12-21 10-54-17

File: bigchaindb/models.py (hunk 1 above)
Comment (ttmc) [reply bigchaindb/models.py]: :+1:

File: docs/server/source/data-models/block-model.rst [hunk 5]
```diff
@@ -9,7 +9,7 @@ A block must contain the following JSON keys
 .. code-block:: json
 
     {
-      "id": "<ID of the block>",
+      "height": "<ID of the block>",
```
Comment (ttmc) [inline docs/server/source/data-models/block-model.rst]: Maybe also change `"<ID of the block>"` to `"<Height of the block>"`?

File: docs/server/source/data-models/block-model.rst [hunk 6]
```diff
@@ -19,7 +19,10 @@ The JSON Keys in a Block
 
 **id**
 
-The block ID denotes the height of the blockchain when the given block was committed.
+The block ``"height"`` (``integer``) denotes the height of the blockchain when the given block was committed.
```
Comment (ttmc) [inline docs/server/source/data-models/block-model.rst]: The title of this section should also be changed, from `**id**` to `**height**`

**Comment by codegeschrei** [comment]
In the docs for `/api/v1/blocks/{block_id}` the response shows `"height": 1,`. When I curl locally I get `"id":1` instead. I think we should make it the same and hint that the id is the height. Otherwise, it is confusing when I expect to get the height and don't see it (or have to guess it).

**Comment by ttmc** [comment]
What will happen to the endpoint

GET /api/v1/blocks?transaction_id={transaction_id}&status={UNDECIDED|VALID|INVALID}

**Comment by kansi** [comment]
The `status` param for the api should be dropped. A block only exists after it has been commited.

**Comment by ttmc** [comment]
> The `status` parameter for the api should be dropped. 

@kansi I guess that will be done in a separate pull request?

**Comment by kansi** [comment]
@ttmc Yes, this PR only implements the blocks api i.e. `GET /api/v1/block/[height]`

**Pull request merged**